File: moldme/Models/Status.cs
namespace moldme.Models;

/// <summary>
/// Represents the status of an entity.
/// </summary>
public enum Status
{
    /// <summary>
    /// The entity is new.
    /// </summary>
    NEW,

    /// <summary>
    /// The entity is in progress.
    /// </summary>
    INPROGRESS,

    /// <summary>
    /// The entity is done.
    /// </summary>
    DONE,

    /// <summary>
    /// The entity is closed.
    /// </summary>
    CLOSED,

    /// <summary>
    /// The entity is canceled.
    /// </summary>
    CANCELED,

    /// <summary>
    /// The entity is pending.
    /// </summary>
    PENDING,

    /// <summary>
    /// The entity is accepted.
    /// </summary>
    ACCEPTED,

    /// <summary>
    /// The entity is denied.
    /// </summary>
    DENIED,
}

File: moldme_frontend/lib/models/status.dart
/// Represents the status of an entity.
enum Status {
  newEntity,
  inProgress,
  done,
  closed,
  canceled,
  pending,
  accepted,
  denied;

  /// Converts an integer from the backend to a Status enum.
  static Status fromInt(int value) {
    return Status.values[value];
  }

  /// Converts a Status enum to an integer for the backend.
  int toInt() {
    return Status.values.indexOf(this);
  }
}